modelscope_agent/agent.py

```diff
@@ -1,377 +1,182 @@
+
+        :param function_list: Optional[List[Union[str, Dict]]] :
+            (1)When str: tool names
+            (2)When Dict: tool cfg
+        :param llm: Optional[Union[Dict, BaseChatModel]]:
+            (1) When Dict: set the config of llm as {'model': '', 'api_key': '', 'model_server': ''}
+            (2) When BaseChatModel: llm is sent by another agent
+        :param storage_path: If not specified otherwise, all data will be stored here in KV pairs by memory
+        :param name: the name of agent
+        :param description: the description of agent, which is used for multi_agent
+        :param instruction: the system instruction of this agent
+        :param kwargs: other potential parameters
         """
-
-        self.llm = llm
-
-        self.agent_type = agent_type
-        self.llm.set_agent_type(agent_type)
-        if issubclass(prompt_generator.__class__, PromptGenerator):
-            self.prompt_generator = prompt_generator
-        else:
-            self.prompt_generator = get_prompt_generator(
-                agent_type, llm, prompt_generator=prompt_generator, **kwargs)
-
-        if issubclass(action_parser.__class__, ActionParser):
-            self.action_parser = action_parser
+        if isinstance(llm, Dict):
+            self.llm_config = llm
+            self.llm = get_chat_model(**self.llm_config)
         else:
-            self.action_parser = get_action_parser(
-                agent_type, llm, action_parser=action_parser, **kwargs)
-
-        self._init_tools(tool_cfg, additional_tool_list)
-
-        if isinstance(tool_retrieval, bool) and tool_retrieval:
-            tool_retrieval = ToolRetrieval()
-        self.tool_retrieval = tool_retrieval
-        if self.tool_retrieval:
-            self.tool_retrieval.construct(
-                [str(t) for t in self.tool_list.values()])
-        self.knowledge_retrieval = knowledge_retrieval
-        self.reset()
-        self.seed = None
-
-    def _init_tools(self,
-                    tool_cfg: Dict = {},
-                    additional_tool_list: Dict = {}):
-        """init tool list of agent. We provide a default tool list, which is initialized by a cfg file.
-        user can also provide user-defined tools by additional_tool_list.
-        The key of additional_tool_list is tool name, and the value is corresponding object.
-
-        Args:
-            tool_cfg (Dict): default tool cfg.
-            additional_tool_list (Dict, optional): user-defined tools. Defaults to {}.
+            self.llm = llm
+        self.stream = True
+
+        self.function_list = []
+        self.function_map = {}
+        if function_list:
+            for function in function_list:
+                self._register_tool(function)
+                try:
+                    self._register_tool(function)
+                except Exception as e:
+                    logger.query_warning(
+                        uuid=kwargs.get('uuid_str', 'local_user'),
```

Comment:
why we need to print uuid_str inside a framework?

Reply:
we should raise the exception and let the upper apps to handle and print the right info

Reply:
removd